native/src/main/java/io/ballerina/lib/sap/Listener.java

```diff
@@ -102,40 +99,26 @@ public static Object immediateStop(BObject client) {
         return stopListener(client);
     }
 
-    @SuppressWarnings("unchecked")
     public static Object detach(BObject listener, BObject service) {
-        ArrayList<BObject> startedServices = (ArrayList<BObject>) listener.getNativeData(
-                SAPConstants.JCO_STARTED_SERVICES);
-        ArrayList<BObject> services = (ArrayList<BObject>) listener.getNativeData(SAPConstants.JCO_SERVICES);
         try {
-            startedServices.remove(service);
-            services.remove(service);
-            return null;
-        } catch (Exception e) {
+            JCoIDocServer server = JCoIDoc.getServer(jcoServerName);
+            listener.addNativeData(SAPConstants.JCO_SERVER, server);
```

Comment:
Shouldn't we remove the server native data when detaching the service?

Reply:
Yes, Changed with [986e899](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/47/commits/986e8992704374cc5b22f3813ebe72a30b144793).